apps/omg_api/lib/block_queue/core.ex

```diff
@@ -203,20 +202,30 @@ defmodule OMG.API.BlockQueue.Core do
 
   defp adjust_gas_price(
          %Core{
+           mined_child_block_num: mined_child_block_num,
+           formed_child_block_num: formed,
+           child_block_interval: block_interval,
+           blocks: blocks,
            parent_height: parent_height,
-           gas_price_adj_params: %GasPriceParams{last_block_mined: {last_parent_height, _mined_block_num}}
+           last_parent_height: last_parent_height
          } = state
-       )
-       when parent_height == last_parent_height,
-       do: state
+       ) do
+    first_blknum = mined_child_block_num + block_interval
+
+    if parent_height == last_parent_height or
```

Comment:
`parent_height <= last_parent_height` to consider back offs. + add a test where you check that a back of doesn't cause gas price update